agimus_controller/agimus_controller/trajectory.py

```diff
@@ -0,0 +1,52 @@
+import numpy as np
+from collections import deque
+from dataclasses import dataclass
+from pinocchio import SE3, Force
+
+
+@dataclass
+class TrajectoryPoint:
+    """Trajectory point aiming at being a reference for the MPC."""
+
+    robot_configuration: np.ndarray
+    robot_velocity: np.ndarray
+    robot_acceleration: np.ndarray
+    robot_effort: np.ndarray
+    forces: dict[Force]  # Dictionary of pinocchio.Force
+    end_effector_poses: dict[SE3]  # Dictionary of pinocchio.SE3
+
+
+class TrajectoryBuffer(deque):
+    """List of variable size in which the HPP trajectory nodes will be."""
+
+    def __init__(self) -> None:
+        super().__init__()
+
+    def add_trajectory_point(self, trajectory_point: TrajectoryPoint):
+        """
+        Add trajectory point to the buffer.
+        """
+        self.append(trajectory_point)
+
+    def get_size(self):
+        """
+        Returns the size of the buffer until the first invalid TrajectoryPoint.
+        """
+        return self.__len__()
+
+    def get_points(self, nb_points: int):
```

Comment:
I think this function needs more work as we need to decide if the buffer should be flushed when results are queried from it or they should stay there and only be overwritten by new data. The other approach is yet another abstract class... Basically, all this comes to the differences on how do we handle cases when we run out of data in buffer.